custom_components/tahoma/sensor.py

```diff
@@ -75,7 +78,15 @@ def unit_of_measurement(self):
         states = self.tahoma_device.active_states
 
         if CORE_TEMPERATURE_STATE in states:
-            # TODO Retrieve core:MeasuredValueType to understand if it is Celsius or Kelvin
+            unit = self.tahoma_device.attributes.get(CORE_MEASURED_VALUE_TYPE)
+            if unit:
```

Comment:
Why not check for Celsius? Now we have a double fallback to Celsius, but it is not super readable in my opinion. I would like it more if it would be something like this.

(I still miss switch/case in Python)

```python
        if CORE_TEMPERATURE_STATE in states:
            measured_value = self.tahoma_device.attributes.get(CORE_MEASURED_VALUE_TYPE)

            if measured_value is not None:
                if "Celcius" in measured_value:
                    return TEMP_CELSIUS

                if "Kelvin" in measured_value:
                    return TEMP_KELVIN

                if "Fahrenheit" in measured_value:
                    return TEMP_FAHRENHEIT       
           
            return TEMP_CELSIUS
```

Reply:
I just thought that way was more pythonesque, but I don't really mind TBH.

Reply:
This is indeed IMO not really readable. You can use a dict to avoid the if else.

Reply:
Do we need to check if the value is in the string? Or can we do a string compare, would be even more readable imo.

```python
if "Celsius" in measured_value:
```

```python
if measured_value == "Celsius":
```

Reply:
`core:TemperatureInCelcius`
I don't have access to Fahrenheit and Kelvin. Not even sure they exist, but I guess it would be the same kind of format.

Reply:
do you like this better:
```
        if CORE_TEMPERATURE_STATE in states:
            units = {
                "core:TemeratureInCelsius": TEMP_CELSIUS,
                "core:TemeratureInKelvin": TEMP_KELVIN,
                "core:TemeratureInFahrenheit": TEMP_FAHRENHEIT,
            }
            return units.get(self.tahoma_device.attributes.get(CORE_MEASURED_VALUE_TYPE), TEMP_CELSIUS)
```

Reply:
Looks better indeed.